infra/testApi.mjs

```diff
@@ -236,6 +181,113 @@ checkLocalErrorNode(localError7.tree, [0, 1],
 
+// MARK: Alternatives endpoint
+const alt_pcontext = [[[14.97651307489794], 0.12711304680349078]]
+const alts_sync = callAlternatives(FPCoreFormula, alt_pcontext, false);
+const alts_async = callAlternatives(FPCoreFormula, alt_pcontext, true);
+assert.deepEqual(alts_sync, alts_async);
+
+async function callAlternatives(fpcore, p_context, async_huh) {
+  const alternatives_json = await getJSONFor('alternatives', 'POST', JSON.stringify({
+    formula: fpcore, sample: p_context
+  }), async_huh);
+  assertIdAndPath(alternatives_json);
+  checkFelids(alternatives_json, ['command', 'alternatives', 'histories', 'derivations', 'splitpoints', 'job', 'path']);
+  assert.equal(Array.isArray(alternatives_json.alternatives), true);
+  return alternatives_json;
+}
+
+// MARK: Cost endpoint
+async function callCost(fpcore, p_context, async_huh) {
+  const cost_json = await getJSONFor('cost', 'POST', JSON.stringify({
+    formula: fpcore, sample: p_context
+  }), async_huh);
+  assertIdAndPath(cost_json);
+  checkFelids(cost_json, ['command', 'cost', 'job', 'path']);
+  assert.equal(cost_json.cost > 0, true);
+  return cost_json;
+}
+
+// MARK: Translate
+async function callTranslate(fpcore, language) {
+  if (language === 'mathjs') {
+    const json = await getJSONFor('mathjs', 'POST', JSON.stringify({ formula: fpcore }), false);
+    checkFelids(json, ['mathjs']);
+    return json;
+  } else {
+    const json = await getJSONFor('translate', 'POST', JSON.stringify(
+      { formula: fpcore, language: language }), false);
+    checkFelids(json, ['language', 'result']);
+    return json;
+  }
+}
+
+// MathJS endpoint
+const test_json = await callTranslate(FPCoreFormula, 'mathjs');
+const expected_value = "sqrt(x + 1.0) - sqrt(x)";
+assert.equal(test_json.mathjs, expected_value);
+
+// Translate endpoint
+const expectedExpressions = {
+  "python": 'def expr(x):\n\treturn math.sqrt((x + 1.0)) - math.sqrt(x)\n',
+  "c": 'double expr(double x) {\n\treturn sqrt((x + 1.0)) - sqrt(x);\n}\n',
+  "fortran": 'real(8) function expr(x)\nuse fmin_fmax_functions\n    real(8), intent (in) :: x\n    expr = sqrt((x + 1.0d0)) - sqrt(x)\nend function\n',
+  "java": 'public static double expr(double x) {\n\treturn Math.sqrt((x + 1.0)) - Math.sqrt(x);\n}\n',
+  "julia": 'function expr(x)\n\treturn Float64(sqrt(Float64(x + 1.0)) - sqrt(x))\nend\n',
+  "matlab": 'function tmp = expr(x)\n\ttmp = sqrt((x + 1.0)) - sqrt(x);\nend\n',
+  "wls": 'expr[x_] := N[(N[Sqrt[N[(x + 1), $MachinePrecision]], $MachinePrecision] - N[Sqrt[x], $MachinePrecision]), $MachinePrecision]\n', // Wolfram 
+  "tex": '\\mathsf{expr}\\left(x\\right) = \\sqrt{x + 1} - \\sqrt{x}\n',
+  "js": 'function expr(x) {\n\treturn Math.sqrt((x + 1.0)) - Math.sqrt(x);\n}\n'
+}
+
+for (const e in expectedExpressions) {
+  const test_json = await callTranslate(FPCoreFormula, e);
+  assert.equal(test_json['language'], e);
+  assert.equal(test_json['result'], expectedExpressions[e]);
+}
```

Comment:
I guess this is fine but if instead of having a `callTranslate` function you just had a `testTranslate` function that took the formula, language, and expected output, then you wouldn't need the for loop and `expectedExpressions` dictionary.